lib/gollum/views/pages.rb

```diff
@@ -41,19 +43,23 @@ def files_folders
               folder_path = @path ? "#{@path}/#{folder}" : folder
               folder_link = %{<li><a href="#{@base_url}/pages/#{folder_path}/" class="folder">#{folder}</a></li>}
 
-              unless folder_links.include?(folder_link)
-                folder_links << folder_link
-
-                folder_link
-              end
+              folders[folder] = folder_link unless folders.key?(folder_link)
             elsif page_path != ".gitkeep"
               if defined? page.format
-                %{<li><a href="#{@base_url}/#{page.escaped_url_path}" class="file">#{page.name}</a></li>}
+                page_link = %{<li><a href="#{@base_url}/#{page.escaped_url_path}" class="file">#{page.name}</a></li>}
+                page_files[page.name] = page_link
               else
-                %{<li><a href="#{@base_url}/#{page.escaped_url_path}#{page.name}" class="file">#{page.name}</a></li>}
+                page_link = %{<li><a href="#{@base_url}/#{page.escaped_url_path}#{page.name}" class="file">#{page.name}</a></li>}
+                page_files[page.name] = page_link
               end
             end
-          }.compact.join("\n")
+          end
+
+          # 1012: All Pages should be rendered as Folders first, then Pages, each sorted alphabetically
+          result = folders.sort_by{ |k, v|  k.downcase }.compact.map{ |f| f[1] }.join("\n") + "\n"
```

Comment:
I think you can accomplish the same with the following, which maybe a little easier to read:
```ruby
result = Hash[folders.sort_by{| key, value | key.downcase} ].values.join("\n") + "\n"
```